File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l=0
        while l<len(numbers)-1:
            r = l+1
            while r<len(numbers):
                if numbers[r]+numbers[l] == target:
                    return [l+1,r+1]
                else:
                    r+=1
            l+=1
        

File: Data Structures & Algorithms/combination-target-sum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @returns {number[][]}
     */
    combinationSum(nums, target) {
        const result = [];
        const current = [];

        function backtrack(startIndex, remaining) {
            if (remaining === 0) {
                result.push([...current]);
                return;
            }

            if (remaining < 0) {
                return;
            }

            for(let i = startIndex; i<nums.length; i++){
                current.push(nums[i])
                backtrack(i, remaining - nums[i])
                current.pop()
            }
        }
        backtrack(0, target);
        return result
    }
}
